waghsk/i2b2-quickstart
# Fix for #11: Make sure of DS_PASS environment variable

Fixes issue [#11](https://github.com/waghsk/i2b2-quickstart/issues/11) by defining datasources in `standalone.xml` instead of separate config files so that environment variable DS_PASS can be directly used in configuration file and don't have to run `sed` command to alter the files.

**Steps to test the fix:**
1. Create `docker-compose.yml` file for testing the fix.
```yaml
version: '3.2'
services:
    i2b2-pg:
        image: i2b2/i2b2-pg:p1
        container_name: i2b2-pg
        ports:
            - "5432:5432"

    i2b2-wildfly:
        container_name: i2b2-wildfly
        build: ./i2b2-wildfly/.
        environment:
            - DS_PASS=newpass
        ports:
            - "8080:8080"
            - "9900:9900"

    i2b2-web:
        container_name: i2b2-web
        image: i2b2/i2b2-web:p1
        command: /run-httpd.sh localhost
        ports:
            - "80:80"
            - "443:443"
```
2. Run `docker-compose up -d i2b2-pg` to bring up the database.
3. Connect to the postgres database and run following SQL statements.
```sql
ALTER USER i2b2 WITH PASSWORD 'newpass' ;
ALTER USER i2b2demodata WITH PASSWORD 'newpass' ;
ALTER USER i2b2hive WITH PASSWORD 'newpass' ;
ALTER USER i2b2imdata WITH PASSWORD 'newpass' ;
ALTER USER i2b2metadata WITH PASSWORD 'newpass' ;
ALTER USER i2b2pm WITH PASSWORD 'newpass' ;
ALTER USER i2b2workdata WITH PASSWORD 'newpass' ;
```
4. Run `docker-compose up i2b2-wildfly i2b2-web` to bring up Apache and Wildfly containers.
5. Browse http://localhost/webclient and verify that you can run a query on patients.

**Limitations:**
This fix has been tested only with Postgresql database. To support other database type (for example: oracle), another `standalone.xml` file might need to be created with proper connection string and any properties specific to that database type.
